pandas/core/indexes/base.py

```diff
@@ -3081,13 +3081,14 @@ def get_value(self, series, key):
         # if we have something that is Index-like, then
         # use this, e.g. DatetimeIndex
         s = getattr(series, '_values', None)
-        if isinstance(s, (ExtensionArray, Index)) and is_scalar(key):
-            try:
-                return s[key]
-            except (IndexError, ValueError):
-
-                # invalid type as an indexer
-                pass
+        if is_scalar(key):
+            if isinstance(s, (Index, ExtensionArray)):
+                try:
+                    iloc = self.get_loc(key)
+                    return s[iloc]
+                except KeyError:
+                    if isinstance(key, (int, np.integer)):
```

Comment:
use is_integer

Reply:
why doesn’t the pass work?

Reply:
needs comments

Reply:
@jreback I've just pushed a new commit that uses is_integer and adds comments.

`pass` doesn't work because we are catching a different exception. So the way I unified the code between `Index` and `ExtensionArray` was by handling the exception differently in the case that the `key` was a scalar.